### ISSUE TITLE
Provide validation of the SBOM content

### ISSUE THREAD

**Issue by sxa** [adoptium/temurin-build#3484]
We have lots of information in our SBOM. As part of SLSA verification requirements we should ensure that as much as possible of the data is validated including checking that the fields with information we populate about the build look "reasonable" and include checking versions of some of the prerequisites and validate that the github SHAs listed are valid.

FYI @andrew-m-leonard @netomi as this came out of the conversation we had a few weeks ago. My first pass of a validation script that does some validation on our nightly JDK17 versions is 
[validateSBOMcontent.sh.txt](https://github.com/adoptium/temurin-build/files/12659266/validateSBOMcontent.sh.txt) - this is very much an initial prototype (and is not currently in git until we decide on a suitable location for it) and will need to be enhanced to ensure that it can handle the correct checks for all of our openjdk versions and probably other checks.

At the moment it checks the GLIBC/GCC and ALSA versions against "expected" versions as well as validating that the GIT shas for the openjdk codebase and temurin-build are present in the corresponding repositories. All suggestions for further checks welcome. Note that this will not currently validate the last GA versions as they have a number of omissions (e.g. no GLIBC etc.) so this can only be reasonably used against the nightly builds at present.

I have been prototyping this by running it as part of a longer build download verification process that validates all the GPG signatures etc. and does other integrity checks on the downloads, so this issue purely covers validation of the SBOM content.

We can also look at [CycloneDX's format validator](https://github.com/CycloneDX/cyclonedx-cli#validate-command) to check that the overall format of the file is as per the CycloneDX specification.

**Comment by netomi** [comment]
ty for the link. I have checked the CycloneDX cli tool and tested it on one of the sbom files. It validates with success while its not clear what happens under the hood.

**Comment by sxa** [comment]
The IVT job is currently working reasonably well, although without a GA build after the [GCC/GLIB fix](https://github.com/adoptium/temurin-build/pull/3447) went in we cannot run this against a GA build. I have put in some fixes to allow it to work with a non-GA build including skipping the verification of the temurin-build SHA since that cannot be easily verified with the current approach unless it was a tagged commit.

This testing has also highlighted the absence of the FreeType version information in the case where the FreeType code is not downloaded - https://github.com/adoptium/temurin-build/issues/3493 has been raised to address this.

At the moment the code is in an `SXAIVT` job on the Eclipse Jenkins server and it performs some of the work itself and calls [validateSBOMcontent.sh.txt](https://github.com/adoptium/temurin-build/files/12793998/validateSBOMcontent.sh.txt) to check the contents of the SBOM file - this will of course be put under source control.

> ty for the link. I have checked the CycloneDX cli tool and tested it on one of the sbom files.

When I saw this I managed to make sure that was integrated into my checks to and it works well as you say - thanks. I'll aim to see what I can do with the other links you've given :-)

**Comment by sxa** [comment]
Going to skip the explicit freetype check for now in the basis of the large number of inconsistent versions discovered in https://github.com/adoptium/temurin-build/issues/3504#issuecomment-1782641566 - will reinstate when that is resolved.
I might try and make sure it's either "N.A." or a real version (regalrdless of what it is) for now

**Comment by sxa** [comment]
I've got the verification now running successfully for all tested fields. It currently runs on a Linux machine and does a series of checks. It does not verify signatures on Windows or Mac builds, but I feel that after some issues we have experienced in the October 2023 release cycle (https://github.com/adoptium/ci-jenkins-pipelines/pull/831, https://github.com/adoptium/ci-jenkins-pipelines/pull/832 and a mac one I don't have a reference to) that will be a requirement going forward, potentially building on the work from @netomi above.

I think I'm probably right in saying that it won't be possible to do that well enough within the job that is running on Linux so we'll need something alongside my existing processes.
